email/Email.php

```diff
@@ -184,103 +204,181 @@ public function setBounceHandlerURL($bounceHandlerURL) {
+	/**
+	 * Attach the specified file to this email message.
+	 *
+	 * @param string $filename Relative or full path to file you wish to attach to this email message.
+	 * @param string|null $attachedFilename Name of the file that should appear once it's sent as a separate attachment.
+	 * @param string|null $mimeType MIME type to use when attaching file. If not provided, will attempt to infer via HTTP::get_mime_type().
+	 * @return $this
+	 */
+	public function attachFile($filename, $attachedFilename = null, $mimeType = null) {
 		if(!$attachedFilename) $attachedFilename = basename($filename);
 		$absoluteFileName = Director::getAbsFile($filename);
 		if(file_exists($absoluteFileName)) {
-			$this->attachFileFromString(file_get_contents($absoluteFileName), $attachedFilename, $mimetype);
+			$this->attachFileFromString(file_get_contents($absoluteFileName), $attachedFilename, $mimeType);
 		} else {
 			user_error("Could not attach '$absoluteFileName' to email. File does not exist.", E_USER_NOTICE);
 		}
 		return $this;
 	}
 
+	/**
+	 * @return string|null
+	 */
 	public function Subject() {
 		return $this->subject;
 	}
 
+	/**
+	 * @return string|null
+	 */
 	public function Body() {
 		return $this->body;
 	}
 
+	/**
+	 * @return string|null
+	 */
 	public function To() {
 		return $this->to;
 	}
 
+	/**
+	 * @return string|null
+	 */
 	public function From() {
 		return $this->from;
 	}
 
+	/**
+	 * @return string|null
+	 */
 	public function Cc() {
 		return $this->cc;
 	}
 
+	/**
+	 * @return string|null
+	 */
 	public function Bcc() {
 		return $this->bcc;
 	}
 
+	/**
+	 * @param string $val
+	 * @return $this
+	 */
 	public function setSubject($val) {
 		$this->subject = $val;
 		return $this;
 	}
 
+	/**
+	 * @param string $val
+	 * @return $this
+	 */
 	public function setBody($val) {
 		$this->body = $val;
 		return $this;
 	}
 
+	/**
+	 * @param string $val
+	 * @return $this
+	 */
 	public function setTo($val) {
 		$this->to = $val;
 		return $this;
 	}
 
+	/**
+	 * @param string $val
+	 * @return $this
+	 */
 	public function setFrom($val) {
 		$this->from = $val;
 		return $this;
 	}
 
+	/**
+	 * @param string $val
+	 * @return $this
+	 */
 	public function setCc($val) {
 		$this->cc = $val;
 		return $this;
 	}
 
+	/**
+	 * @param string $val
+	 * @return $this
+	 */
 	public function setBcc($val) {
 		$this->bcc = $val;
 		return $this;
 	}
 
 	/**
 	 * Set the "Reply-To" header with an email address.
-	 * @param string $email The email address of the "Reply-To" header
+	 *
+	 * @param string $val
+	 * @return $this
+	 */
+	public function setReplyTo($val) {
+		$this->addCustomHeader('Reply-To', $val);
+		return $this;
+	}
+
+	/**
+	 * @param string $email
+	 * @return $this
+	 * @deprecated 3.5 Use the "setReplyTo" method instead
 	 */
 	public function replyTo($email) {
+		Deprecation::notice('4.0', 'Use the "setReplyTo" method instead');
 		$this->addCustomHeader('Reply-To', $email);
 		return $this;
```

Comment:
Not just DRY, but also "dogfooding" it's own deprecation notice, hah. Not sure how I missed that, so thanks for pointing it out. Fixed in my latest push.